### PR TITLE
Fix (or try to) Pose3Pose3XYYaw

### PR DESCRIPTION
**Pull request by Affie**

### CODE REVIEW

File: src/factors/PartialPose3.jl [hunk 1]
```diff
@@ -113,7 +113,29 @@ getManifold(::Pose3Pose3XYYaw) = SpecialEuclidean(2)
 ## NOTE, Yaw only works if you assume a preordained global reference point, such as identity_element(Pose3)
 function (cfo::CalcFactor{<:Pose3Pose3XYYaw})(X, wTp, wTq )
   #
+  #FIXME if else is for comparing 2 different aprouches in factor
+  if true
+    
+  M = SpecialEuclidean(2)
 
+  rx = normalize(wTp.parts[2][1:2, 1])
```
Comment (dehann) [inline src/factors/PartialPose3.jl:121]: https://github.com/JuliaRobotics/RoME.jl/commit/85a997fc37973b71a5d20f3b729e40f984bb98bf#r57341417

File: test/testPartialPose3.jl [hunk 2]
```diff
@@ -332,7 +362,8 @@ end
 @test size(val, 2) == N
 
 estmu1mean = Statistics.mean(val[collect(DFG.getSolverData(f1).fnc.usrfnc!.partial),:],dims=2)
-estmu2mean = Statistics.mean(val[collect(DFG.getSolverData(f2).fnc.usrfnc!.partial),:],dims=2)
+# estmu2mean = Statistics.mean(val[collect(DFG.getSolverData(f2).fnc.usrfnc!.partial),:],dims=2)
+estmu2mean = Statistics.mean(val[[1,2,6],:],dims=2)
```
Comment (dehann) [inline test/testPartialPose3.jl:366]: https://github.com/JuliaRobotics/RoME.jl/commit/85a997fc37973b71a5d20f3b729e40f984bb98bf#r57342502

File: test/testPartialPose3.jl [hunk 3]
```diff
@@ -379,7 +410,7 @@ testsMeasurements_xyz_rpy = [
   [10.,  0,  0, 0, 0, ψ],
   [ 0., 10,  0, 0, 0, ψ],
   [ 0.,  0, 10, 0, 0, ψ],
-  [ 0.,  0, 10, 0, 0, ψ],
+  [ 0., 15, 10, 0, 0, ψ],
```
Comment (dehann) [inline test/testPartialPose3.jl:413]: https://github.com/JuliaRobotics/RoME.jl/commit/85a997fc37973b71a5d20f3b729e40f984bb98bf#r57342525

**Comment by Affie** [comment]
I've tried to run a solve with 5 variables to help verify the factor and the solve is extremely slow. 
The factor itself works with my tests, but we should confirm that the tests are actually correct.

File: src/factors/PartialPose3.jl (hunk 1 above)
Comment (dehann) [reply src/factors/PartialPose3.jl:121]: Hi Johan, did you see my comment on why i think there is a problem with sub-selecting from rotation matrix?  This does not give just yaw, pitch is also mixed in on the first column.

File: src/factors/PartialPose3.jl [hunk 4]
```diff
@@ -113,7 +113,29 @@ getManifold(::Pose3Pose3XYYaw) = SpecialEuclidean(2)
 ## NOTE, Yaw only works if you assume a preordained global reference point, such as identity_element(Pose3)
 function (cfo::CalcFactor{<:Pose3Pose3XYYaw})(X, wTp, wTq )
   #
+  #FIXME if else is for comparing 2 different aprouches in factor
```
Comment (dehann) [inline src/factors/PartialPose3.jl:116]:
```suggestion
  #FIXME if else is for comparing 2 different approaches in factor
```

**Pull request merged**